File: 250225/각 자리 숫자의 제곱/square-of-each-digit.java
import java.io.*;
import java.util.*;

public class Main {
    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static int n;

    public static int getSquareTotal(int n) {
        if (n < 10)
            return n * n;
        int digit = n % 10;
        return getSquareTotal(n - 1) + d * d;
    }

    public static void main(String[] args) throws IOException {
        n = Integer.parseInt(br.readLine());
        System.out.println(getSquareTotal(n));
    }
}

File: 250225/각 자리 숫자의 제곱/square-of-each-digit.cpp
#include <iostream>

using namespace std;

int get_square_total(int n) {
    if (n < 10)
        return n * n;
    int digit = n % 10;
    return get_square_total(n / 10) + digit * digit;
}

int main() {
    int n;
    cin >> n;

    cout << get_square_total(n);
    return 0;
}